apps/test/unit/tutorialExplorer/tutorialSetTest.jsx

```diff
@@ -0,0 +1,216 @@
+import React from 'react';
+import {shallow} from 'enzyme';
+import {expect} from '../../util/configuredChai';
+import Tutorial from '@cdo/apps/tutorialExplorer/tutorial';
+import TutorialSet from '@cdo/apps/tutorialExplorer/tutorialSet';
+import TutorialDetail from '@cdo/apps/tutorialExplorer/tutorialDetail';
+
+const TUTORIAL_1 = {
+  name: 'Tutorial 1',
+  code: 'tutorial-1',
+};
+
+const TUTORIAL_2 = {
+  name: 'Tutorial 2',
+  code: 'tutorial-2',
+};
+
+const DEFAULT_PROPS = {
+  tutorials: [],
+  filters: {},
+  localeEnglish: true,
+  disabledTutorials: [],
+};
+
+describe('TutorialSet', () => {
+  it('renders empty set of tutorials', () => {
+    const wrapper = shallow(
+      <TutorialSet {...DEFAULT_PROPS}/>
+    );
+    const instance = wrapper.instance();
+    expect(wrapper).to.containMatchingElement(
+      <div>
+        <TutorialDetail
+          showing={false}
+          item={null}
+          closeClicked={instance.tutorialDetailClosed}
+          localeEnglish={true}
+          disabledTutorial={false}
+          changeTutorial={instance.changeTutorial}
+        />
+        <div>
+          Your search returned no results.
```

Comment:
I find it somewhat strange to have tests duplicate strings from the app itself.  It feels like robustness through redundant duplication rather than a test of code working.  (Similarly for checking of CSS properties that have shown up in other React tests in recent months.)

Reply:
Yeah, I'm a little torn about this.

Having the tests import localization and using it here instead of the string itself is probably the right way to go.  It means the test won't break if the localizations change or a different locale is selected, and as you say it may isolate the test's intent from other factors.

That said, I sort of like seeing the "plain english" output - it feels like it documents the intended render output better.  I suppose that's what storybook is for though.